### PR TITLE
Update travis.yml + fix tests

### PR DESCRIPTION
**Pull request by jayeff**
* use latest Crate Version
* test more ruby versions
* fix broken test
* remove deprecation warnings

### CODE REVIEW

File: spec/test_server.rb [hunk 1]
```diff
@@ -20,44 +21,40 @@
...
   NAME = "TestCluster"
+  HOST = "127.0.0.1"
+  TIMEOUT = 10
 
-
-  def initialize(crate_home = nil, port = nil, host = "127.0.0.1")
-    @crate_home = crate_home || CRATE_PATH
-    @port = port || TEST_PORT
-    @host = host
+  def initialize(crate_home = '~/crate', port = 4209, run_in_background = false)
```
Comment (chaudum) [inline spec/test_server.rb]: we use test port 44200 in python.

File: spec/test_server.rb [hunk 2]
```diff
@@ -20,44 +21,40 @@
...
+    @crate_home = crate_home
+    @port = port
+    @run_in_background = run_in_background
   end
 
   def start
-    cmd = "sh #{CRATE_PATH}/bin/crate #{start_params}"
-    @pid = spawn(cmd, :out => "/tmp/crate_test_server.out", :err => "/tmp/crate_test_server.err")
+    cmd = "sh #{File.join(@crate_home, 'bin/crate')} #{start_params}"
```
Comment (chaudum) [inline spec/test_server.rb]: File.join(@crate_home, 'bin', 'crate') ?

File: spec/test_server.rb [hunk 3]
```diff
@@ -20,44 +21,40 @@
...
+    @pid = spawn(cmd, out: "/tmp/crate_test_server.out",
+                 err: "/tmp/crate_test_server.err")
     Process.detach(@pid)
-    puts 'starting'
+    puts 'Starting Crate... (this will take a few seconds)'
     time_slept = 0
```
Comment (chaudum) [inline spec/test_server.rb]: `interval = 2`

File: spec/test_server.rb [hunk 4]
```diff
@@ -20,44 +21,40 @@
...
     while true
-      puts "Crate not yet fully available. Waiting since #{time_slept} seconds..." unless alive?
+      if !alive? and time_slept > TIMEOUT
+        puts "Crate hasn't started for #{TIMEOUT} seconds. Giving up now..."
+        exit
+      end
+      if alive? and @run_in_background
+        exit
+      end
       sleep(2)
```
Comment (chaudum) [inline spec/test_server.rb]: `sleep(interval)`

**Comment by jayeff** [comment]
@chaudum Updated. see 5281943

**Comment by chaudum** [comment]
ack
pls squash commit so I can merge

**Comment by chaudum** [comment]
could you pls add a changelog entry too.

**Comment by jayeff** [comment]
Changelog Entry added

**Pull request merged**